rust/core-lib/src/event_context.rs

```diff
@@ -494,6 +495,25 @@ impl<'a> EventContext<'a> {
         self.editor.borrow_mut().dec_revs_in_flight();
     }
 
+    /// Checks a newline needs to be added to the end of the file
+    pub(crate) fn pre_save_newline(&mut self) -> bool {
+        let editor = self.editor.borrow();
+        let ed_text = editor.get_buffer();
+        let view = self.view.borrow();
+        let offset = view.line_of_offset(ed_text, ed_text.len()) + 1;
```

Comment:
this name is slightly confusing; we use `offset` to refer to byte offsets, and this is a line number.